internal/features/stacks/jobs/schema_test.go

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package jobs
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"io/fs"
+	"log"
+	"path/filepath"
+	"sync"
+	"testing"
+	"testing/fstest"
+
+	"github.com/hashicorp/go-version"
+	lsctx "github.com/hashicorp/terraform-ls/internal/context"
+	"github.com/hashicorp/terraform-ls/internal/document"
+	"github.com/hashicorp/terraform-ls/internal/features/stacks/state"
+	"github.com/hashicorp/terraform-ls/internal/job"
+	globalState "github.com/hashicorp/terraform-ls/internal/state"
+	tfaddr "github.com/hashicorp/terraform-registry-address"
+)
+
+func TestPreloadEmbeddedSchema_basic(t *testing.T) {
+	ctx := context.Background()
+	dataDir := "data"
+	schemasFS := fstest.MapFS{
+		dataDir:                            &fstest.MapFile{Mode: fs.ModeDir},
+		dataDir + "/registry.terraform.io": &fstest.MapFile{Mode: fs.ModeDir},
+		dataDir + "/registry.terraform.io/hashicorp":              &fstest.MapFile{Mode: fs.ModeDir},
+		dataDir + "/registry.terraform.io/hashicorp/random":       &fstest.MapFile{Mode: fs.ModeDir},
+		dataDir + "/registry.terraform.io/hashicorp/random/1.0.0": &fstest.MapFile{Mode: fs.ModeDir},
+		dataDir + "/registry.terraform.io/hashicorp/random/1.0.0/schema.json.gz": &fstest.MapFile{
+			Data: gzipCompressBytes(t, []byte(randomSchemaJSON)),
+		},
+	}
+
+	gs, err := globalState.NewStateStore()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ss, err := state.NewStackStore(gs.ChangeStore, gs.ProviderSchemas)
+	if err != nil {
+		t.Fatal(err)
+	}
+	stackPath := "teststack"
+
+	cfgFS := fstest.MapFS{
+		// These are somewhat awkward double entries
+		// to account for io/fs and our own path separator differences
+		// See https://github.com/hashicorp/terraform-ls/issues/1025
+		stackPath + "/providers.tfstack.hcl": &fstest.MapFile{
+			Data: []byte{},
+		},
+		filepath.Join(stackPath, "providers.tfstack.hcl"): &fstest.MapFile{
+			Data: []byte(`required_providers {
+	random = {
+		source = "hashicorp/random"
+		version = "1.0.0"
+	}
+}
+`),
```

Comment:
Why do they have different contents? Is the first one only used to check if something exists but not actually read?

Reply:
Yeah, without the first entry, listing the files in `stackPath` will fail on Windows:
![Screenshot 2024-11-28 170616](https://github.com/user-attachments/assets/a9ed75b3-5bdf-4028-bb84-2a0588328580)

With both entries:
![Screenshot 2024-11-28 170806](https://github.com/user-attachments/assets/c0a87489-739d-4727-852f-ac6c95c2b331)